app/jobs/application_job.rb

```diff
@@ -4,4 +4,17 @@ class ApplicationJob < ActiveJob::Base
   def self.discard_and_log(exception)
     discard_on(exception) { |_, error| Rails.logger.warn(error) }
   end
+
+  def run_exclusively(transaction: true, lock_name: self.class.name)
+    name = "content-publisher:#{lock_name}"
```

Comment:
After our face-to-face chat I don't think this needs a change - `#snakecase` is a method from a library rather than a rails method. We could use `#underscore` but that seems a bit strange with slashes between namespaces.